File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    var answer = [];
    
    let pattern1 = [1, 2, 3, 4, 5];
    let pattern2 = [2, 1, 2, 3, 2, 4, 2, 5];
    let pattern3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    
    let scores = [0, 0, 0];
    
    for(let i = 0; i<answers.length;i++) {
        if(pattern1[i % pattern1.length] == answers[i]) {
            scores[0] ++;
        }
        if(pattern2[i % pattern2.length] == answers[i]) {
            scores[1] ++;
        }
        if(pattern3[i % pattern3.length] == answers[i]) {
            scores[2] ++;
        }
    }
    let maxScore = Math.max(...scores);
    for(let j = 0;j<scores.length;j++) {
        if(scores[j] == maxScore)
        {
            answer.push(j+1);
        }
    }
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    pattern1 = [1,2,3,4,5]
    pattern2 = [2,1,2,3,2,4,2,5]
    pattern3 = [3,3,1,1,2,2,4,4,5,5]
    
    pattern1_count = 0
    pattern2_count = 0
    pattern3_count = 0
    for i in range(len(answers)):
        if pattern1[i%len(pattern1)] == answers[i]:
            pattern1_count+=1
        if pattern2[i%len(pattern2)] == answers[i]:
            pattern2_count+=1
        if pattern3[i%len(pattern3)] == answers[i]:
            pattern3_count+=1
        
    max_score = max(pattern1_count, pattern2_count, pattern3_count)
    if pattern1_count == max_score: answer.append(1)
    if pattern2_count == max_score: answer.append(2)
    if pattern3_count == max_score: answer.append(3)
    return answer